src/beast/beast/module/core/files/File.cpp

```diff
@@ -28,23 +28,11 @@
 
 namespace beast {
 
-// We need to make a shared singleton or else there are
-// issues with the leak detector and order of detruction.
-//
-class NonexistentHolder
-{
-public:
-    static NonexistentHolder* getInstance()
-    {
-        return SharedSingleton <NonexistentHolder>::getInstance();
-    }
-
-    File file;
-};
+static File s_nonexistent;
 
-File const& File::nonexistent ()
+File const& File::nonexistent()
 {
-    return NonexistentHolder::getInstance ()->file;
+    return s_nonexistent;
```

Comment:
@HowardHinnant After thinking about it, I agree with you, we should move it inside the function. Even if I show order of initialization isn't a problem now, we left a potential problem for the future.